adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -307,37 +307,37 @@ void handleCredentialsResult(final AsyncResult<String> attempt, final MqttEndpoi
             });
 
             endpoint.closeHandler(v -> {
-                LOG.debug("connection closed with tenantId [{}], authId [{}], deviceId [{}]",
+                LOG.debug("connection closed to device (tenantId: {}, authId: {}, deviceId: {})",
                         authObject.getTenantId(), authObject.getAuthId(), deviceId);
                 if (registrationAssertions.remove(endpoint) != null)
-                    LOG.trace("removed registration assertion for tenantId [{}], authId [{}], deviceId [{}]",
+                    LOG.trace("removed registration assertion for device (tenantId: {}, authId: {}, deviceId: {})",
                             authObject.getTenantId(), authObject.getAuthId(), deviceId);
             });
 
         } else {
-            LOG.debug("authentication failed for tenantId [{}], authId [{}] rejected [{}]",
+            LOG.debug("authentication failed for device (tenantId: {}, authId: {} rejected: {})",
                     authObject.getTenantId(), authObject.getAuthId(), MqttConnectReturnCode.CONNECTION_REFUSED_NOT_AUTHORIZED);
             endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_NOT_AUTHORIZED);
         }
     }
 
     private void publishMessage(final MqttEndpoint endpoint, final String tenantId, final String deviceId,
             final MqttPublishMessage message, final ResourceIdentifier resource) {
-        LOG.trace("received message [tenantId [{}], deviceId [{}], topic: [{}], QoS: [{}], payload [{}]]", tenantId, deviceId,
+        LOG.trace("received message for device (tenantId: {}, deviceId: {}, topic: {}, QoS: {}, payload: [{}])", tenantId, deviceId,
```

Comment:
I think we cannot assume the payload to be a string here. In any case, we should at least put this log statement into an if  block that is only executed if the logger is actually configured for trace level. Otherwise we always try to unmarshal the payload into a string, regardless of whether it actually gets logged or not. IMHO we should not log the payload anyway, because we might also open up a door for attackers to actually wreck our log ... the payload size should be enough FMPOV.

Reply:
Full ack: I will remove the payload log.